pkg/webhook/preflight/nutanix/failuredomain.go

```diff
@@ -0,0 +1,205 @@
+// Copyright 2025 Nutanix. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package nutanix
+
+import (
+	"context"
+	"fmt"
+
+	netv4 "github.com/nutanix/ntnx-api-golang-clients/networking-go-client/v4/models/networking/v4/config"
+	"k8s.io/apimachinery/pkg/api/errors"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	capxv1 "github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/api/external/github.com/nutanix-cloud-native/cluster-api-provider-nutanix/api/v1beta1"
+	"github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pkg/webhook/preflight"
+)
+
+type failureDomainCheck struct {
+	failureDomainName string
+	namespace         string
+	field             string
+	kclient           ctrlclient.Client
+	nclient           client
+}
+
+func (fdc *failureDomainCheck) Name() string {
+	return "NutanixFailureDomains"
```

Comment:
Since we have a check for every use of a failure domain, let's use the singular in the name.

```suggestion
	return "NutanixFailureDomain"
```